### PR TITLE
Add shared workflows

### PR DESCRIPTION
**Pull request by bsmth**
This PR is aligning this repository for organization conventions.

See https://github.com/mdn/mdn/issues/309 for more details.

### CODE REVIEW

**Comment by rebloor** [comment]
@bsmth Is there a way I can test these changes?

File: .prettierrc.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "bracketSameLine": true
```
[the remark this answers is not in the document]
Comment (bsmth) [reply .prettierrc.json:2]: At the moment, copying over from https://github.com/mdn/content/blob/main/.prettierrc.json, I'm not sure what the motivation for it is, though

[the remark this answers is not in the document]
Comment (schalkneethling) [reply .prettierrc.json:2]: > At the moment, copying over from https://github.com/mdn/content/blob/main/.prettierrc.json, I'm not sure what the motivation for it is, though

My recommendation would be, if we are not already enforcing a Prettier standard, let's go with the defaults.

[the remark this answers is not in the document]
Comment (bsmth) [reply .prettierrc.json]: OK, sounds good. Reset in https://github.com/mdn/webextensions-examples/pull/508/commits/71ed91ff2411e5c0b506fbbf6234df7e6e5d4dbe

**Comment by schalkneethling** [comment]
> @bsmth Is there a way I can test these changes?

Hey! So, for the most part, these changes should be transparent to everyone. In terms of the actions, CodeQL will ensure our code is secure, prevent things like accidentally leaking credentials, and help prevent people from sneaking in malicious code (it might be less of a problem here, but we are adding it everywhere code is written).

The idle workflow will mark issues and pull requests as idle, indicating to maintainers that it needs attention. The welcome action does as it states on the tin :) It will add a comment welcoming folks to the repo and guiding them on what to expect and other ways of contributing.

**Comment by schalkneethling** [comment]
@bsmth I think we should add a `CODEOWNERS` here and add `core-content` and @rebloor if they are OK with it. We can also do it in a separate PR if you prefer. Let me know.

**Review by schalkneethling — APPROVED** [review]

**Comment by bsmth** [comment]
Thanks @schalkneethling, merging shortly. I'll do a separate PR for the codeowners

**Pull request merged**